app/src/main/java/com/github/libretube/ui/fragments/ChannelContentFragment.kt

```diff
@@ -0,0 +1,182 @@
+        super.onConfigurationChanged(newConfig)
+        // manually restore the recyclerview state due to https://github.com/material-components/material-components-android/issues/3473
+        binding.channelRecView.layoutManager?.onRestoreInstanceState(recyclerViewState)
+    }
+
+    private fun loadChannelTab(tab: ChannelTab) = lifecycleScope.launch {
+        val response = try {
+            withContext(Dispatchers.IO) {
+                RetrofitInstance.api.getChannelTab(tab.data)
+            }.apply {
+                content = content.deArrow()
+            }
+        } catch (e: Exception) {
+            binding.progressBar.visibility = View.GONE
+            return@launch
+        }
+        withContext(Dispatchers.Main) {
+            nextPage = response.nextpage
+        }
+        val binding = _binding ?: return@launch
+        searchChannelAdapter = SearchChannelAdapter()
+        binding.channelRecView.adapter = searchChannelAdapter
+
+        searchChannelAdapter?.submitList(response.content)
+        binding.progressBar.visibility = View.GONE
+        binding.channelRecView.visibility = View.VISIBLE
+        isLoading = false
+    }
+
+    private fun loadNextPage(isVideo: Boolean, tab: ChannelTab) = lifecycleScope.launch {
+        try {
+            isLoading = true
+            if (isVideo) {
+                fetchChannelNextPage(nextPage ?: return@launch).let {
+                    nextPage = it
+                }
+            } else {
+                fetchTabNextPage(nextPage ?: return@launch, tab).let {
+                    nextPage = it
+                }
+            }
+
+        } catch (e: Exception) {
+            isLoading = false
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val tabData = try {
+            Json.decodeFromString<ChannelTab>(arguments?.getString(IntentData.tabData)?:"")
+        } catch (e: Exception) {
+            null
+        }
+
+        channelId = arguments?.getString(IntentData.channelId)
+        nextPage = arguments?.getString(IntentData.nextPage)
+
+        binding.channelRecView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                super.onScrollStateChanged(recyclerView, newState)
+                recyclerViewState = binding.channelRecView.layoutManager?.onSaveInstanceState()
+            }
+
+            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                val visibleItemCount: Int = recyclerView.layoutManager!!.childCount
+                val totalItemCount: Int = recyclerView.layoutManager!!.getItemCount()
+                val firstVisibleItemPosition: Int =
+                    (recyclerView.layoutManager as LinearLayoutManager).findFirstVisibleItemPosition()
```

Comment:
```suggestion
                val visibleItemCount = recyclerView.layoutManager!!.childCount
                val totalItemCount = recyclerView.layoutManager!!.getItemCount()
                val firstVisibleItemPosition =
                    (recyclerView.layoutManager as LinearLayoutManager).findFirstVisibleItemPosition()
```

Reply:
Honestly I copy pasted this code hehe that's why I forgot to refactor it because your code for bottom reached listener was misbehaving

```
fun RecyclerView.addOnBottomReachedListener(onBottomReached: () -> Unit) {
    viewTreeObserver.addOnScrollChangedListener {
        if (!canScrollVertically(1)) onBottomReached()
    }
}
```

is misbehaving on second time you use it, instead of triggering when reached on bottom. It is triggered on each scroll. Atleast that was happening in my case. you might wanna check it out.